### DISCUSSION TITLE
The pupil diameter and difference between stereo-mode and the single-mode

### DISCUSSION THREAD

**Discussion by QuinzyW** [openPupil/Open-PupilEXT#14] [Q&A] [answered]
Thanks for your Open source spirit, We could use the PupilEXT to collect the pupil diameter of the subject. But there are still some questions need to communicate with you.
1. the diameter showed in the PupilEXT
- when we use a 5mm diameter circle to test, it shows 100 in both the stereo-mode and the single-mode. So what's the Unit of 100(μm?) and why 100? I couldn't understand.
- We also use a 14mm diameter circle to test and get 240.
2. The difference between the stereo-mode and the single-mode
- when we make the test circle move back and forth, the diameter changes. So is it right that the stereo-mode and the single-mode, after calibration, work in the same way, and the stereo-mode can additionally provide a mean diameter?

**Comment by QuinzyW** [comment]
We also use a 14mm diameter circle to test and get 280.(sorry a error)

**Comment by QuinzyW** [comment]
> Thanks for your Open source spirit, We could use the PupilEXT to collect the pupil diameter of the subject. But there are still some questions need to communicate with you.
> 
> 1. the diameter showed in the PupilEXT
> 
> * when we use a 5mm diameter circle to test, it shows 100 in both the stereo-mode and the single-mode. So what's the Unit of 100(μm?) and why 100? I couldn't understand.
> * We also use a 14mm diameter circle to test and get 240.
> 
> 1. The difference between the stereo-mode and the single-mode
> 
> * when we make the test circle move back and forth, the diameter changes. So is it right that the stereo-mode and the single-mode, after calibration, work in the same way, and the stereo-mode can additionally provide a mean diameter?

We also use a 14mm diameter circle to test and get `280`.(sorry a mistake)

**Comment by BZandi** [comment, marked as answer]
In stereo mode, you have two cameras and the pupil diameter from image0 (camera0) and image1 (camera1) is provided in pixel quantity, respectively. Then the physical pupil diameter (in mm) is calculated using stereo matching. For further details on the stereo camera measurement, please read our paper: https://www.frontiersin.org/articles/10.3389/fnins.2021.676220/full

Here is an example of how the CSV log file looks when recording a circular reference object in stereo mode: https://github.com/openPupil/Open-PupilEXT/blob/main/Misc/Matlab_Pupil_Data_Evaluation/00_Data/Pupil_Log_Data/BZ_Pupillendaten_Pure_ROI03_Referenz_X.csv

As you can see, two diameter values are provided in pixel in the CSV file ( column: `diameter[px]`, column `diameterSec[px]`). The `physicaldiameter[mm]` coloumn provides the pupil diameter in mm, which is calculated using stereo matching. 

However, I suspect that the values you reported in your thread are not from the CSV log file but from the data table in the GUI. In the GUI's data table, you need to read the `Pupil physical diameter` row, which states the pupil diameter in mm. The `pupil diameter` row in the GUI's data table is the size in pixel. See this video, for example, where you can see the different data table rows and the `Pupil physical diameter` value:
https://github.com/openPupil/Open-PupilEXT/blob/main/Misc/img/PupilEXT-Introduction.gif

 If the value of the diameter changes with different distances to the cameras, it has to do with the fact that the value is given in pixels and the number naturally changes when the distance to the camera is changed, but the `Pupil physical diameter` row should be approximately steady, as it is provided in mm.

Can you please provide a complete screenshot of the PupilEXT GUI while you track your reference object in stereo mode. Make sure that the data table is open in PupilEXT when you capture the screenshot. Maybe you read the wrong row in the data table, or something went wrong during your stereo calibration.

**Comment by QuinzyW** [comment]
Yes, I mix `Pupil physical diameter` and `Pupil diameter`. The problem has been solved. Thanks.
